runelite-client/src/main/java/net/runelite/client/plugins/kourendlibrary/KourendLibraryTutorialOverlay.java

```diff
@@ -0,0 +1,109 @@
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.kourendlibrary;
+
+import net.runelite.api.Client;
+import net.runelite.api.Player;
+import net.runelite.api.coords.WorldPoint;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.OverlayPriority;
+import net.runelite.client.ui.overlay.components.LineComponent;
+import net.runelite.client.ui.overlay.components.PanelComponent;
+import javax.inject.Inject;
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+
+public class KourendLibraryTutorialOverlay extends Overlay
+{
+	private final Client client;
+	private final KourendLibraryConfig config;
+	private final Library library;
+
+	private final PanelComponent panelComponent;
+	private final LineComponent noDataMessageComponent;
+	private final LineComponent incompleteMessageComponent;
+	private final LineComponent completeMessageComponent;
+	private final LineComponent sidebarMessageComponent;
+
+	@Inject
+	private KourendLibraryTutorialOverlay(Client client, KourendLibraryConfig config, Library library)
+	{
+		this.client = client;
+		this.config = config;
+		this.library = library;
+
+		panelComponent = new PanelComponent();
+		panelComponent.setPreferredSize(new Dimension(177, 0));
+
+		noDataMessageComponent = LineComponent.builder().left("Click on the white squares to start finding books.").build();
+		incompleteMessageComponent = LineComponent.builder().left("Some books have been found. Keep checking marked bookcases to find more.").build();
+		completeMessageComponent = LineComponent.builder().left("All books found.").build();
+		sidebarMessageComponent = LineComponent.builder().left("Locations are in the sidebar.").build();
+
+		setPriority(OverlayPriority.LOW);
+		setPosition(OverlayPosition.TOP_LEFT);
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+		if (!config.showTutorialOverlay())
+		{
+			return null;
+		}
+
+		Player player = client.getLocalPlayer();
+		if (player == null)
+		{
+			return null;
+		}
+
+		WorldPoint playerLoc = player.getWorldLocation();
+		if (playerLoc.getRegionID() != KourendLibraryPlugin.REGION)
```

Comment:
this should probably mirror the other overlay's hidden field

Reply:
I'm not sure why the current overlay even has a hidden flag, since Overlays that have been removed don't render. 
It also leads to a weird situation where If the user starts the client with the plugin enabled, they see the marks, but if they toggle the plugin even once before going to the library, they don't see the marks until they start trying to find the books.

Reply:
idr exactly why we went with a flag instead of adding/removing the overlay, but the behavior was to only show the overlay when you actually start interacting with the minigame. I don't want this to say "interact with the white squares" when there are no white squares because you haven't interacted yet.